storage/remote/queue_manager_test.go

```diff
@@ -2623,6 +2625,168 @@ func TestPopulateV2TimeSeries_MetadataAndTypeAndUnit(t *testing.T) {
 	}
 }
 
+// TestPopulateV2TimeSeries_ScrapeCacheFallback tests that when metadata.send=true
+// and WAL metadata is missing, RW2 looks up metadata from the scrape cache.
+func TestPopulateV2TimeSeries_ScrapeCacheFallback(t *testing.T) {
+	metadataStore := &TestMetaStore{
+		Metadata: []scrape.MetricMetadata{
+			{
+				MetricFamily: "test_gauge",
+				Type:         model.MetricTypeGauge,
+				Help:         "Test gauge from scrape cache",
+				Unit:         "bytes",
+			},
+			{
+				MetricFamily: "test_counter_total",
+				Type:         model.MetricTypeCounter,
+				Help:         "Test counter from scrape cache",
+				Unit:         "seconds",
+			},
+		},
+	}
+
+	target := &scrape.Target{}
+	target.SetMetadataStore(metadataStore)
+
+	fakeManager := &fakeManager{
+		activeTargets: map[string][]*scrape.Target{
+			"test_job": {target},
+		},
+	}
+
+	scrapeMgr := &scrapeManagerMock{
+		ready: true,
+	}
+
+	metadataWatcher := NewMetadataWatcher(promslog.NewNopLogger(), scrapeMgr, "test", nil, interval, deadline)
+	metadataWatcher.manager = fakeManager
+
+	tests := []struct {
+		name                  string
+		metricName            string
+		walMetadata           *metadata.Metadata
+		metadataWatcher       *MetadataWatcher
+		enableTypeAndUnit     bool
+		expectedHelp          string
+		expectedType          writev2.Metadata_MetricType
+		expectedUnit          string
+		expectedMetadataCount int
+	}{
+		{
+			name:                  "no WAL, scrape cache provides all metadata",
+			metricName:            "test_gauge",
+			walMetadata:           nil,
+			metadataWatcher:       metadataWatcher,
+			enableTypeAndUnit:     false,
+			expectedHelp:          "Test gauge from scrape cache",
+			expectedType:          writev2.Metadata_METRIC_TYPE_GAUGE,
+			expectedUnit:          "bytes",
+			expectedMetadataCount: 1,
+		},
+		{
+			name:       "WAL metadata exists, scrape cache not consulted",
+			metricName: "test_gauge",
```

Comment:
Let's add more complex metrics like histogram and maybe OpenMetrics 2.0 counters (`foo` vs `foo_total`) to exercise tricky cases.